quantum/process_keycode/process_autocorrect.c

```diff
@@ -19,6 +19,30 @@
 #    include "autocorrect_data_default.h"
 #endif
 
+#ifndef AUTOCORRECT_LINK_BYTE_COUNT
+#    define AUTOCORRECT_LINK_BYTE_COUNT 2
+#endif
+
+#if AUTOCORRECT_LINK_BYTE_COUNT < 2 || AUTOCORRECT_LINK_BYTE_COUNT > 3
+#    error "AUTOCORRECT_LINK_BYTE_COUNT must be either 2 or 3 bytes."
+#endif
+
+#if defined(__AVR__) && DICTIONARY_SIZE > UINT16_MAX
+#    define AUTOCORRECT_PGM_READ_BYTE(offset) pgm_read_byte_far((uint32_t)autocorrect_data + (offset))
```

Comment:
From what I understand, the far function is really only needed for avr devices that have more than 64k of flash. Considering that qmk only supports one controller that supports that (and is incredibly uncommon, at this point, given STM32 and RP2040 controllers are cheaper)...  is this really necessary?